src/chain.h

```diff
@@ -429,10 +430,11 @@ class CBlockIndex
 
     std::string ToString() const
     {
-        return strprintf("CBlockIndex(pprev=%p, nHeight=%d, merkle=%s, hashBlock=%s)",
+        return strprintf("CBlockIndex(pprev=%p, nHeight=%d, merkle=%s, hashBlock=%s, HasSolution=%s)",
             pprev, nHeight,
             hashMerkleRoot.ToString(),
-            GetBlockHash().ToString());
+            phashBlock ? GetBlockHash().ToString() : "(nil)",
+            HasSolution());
```

Comment:
Note that tinyformat correctly prints "true" or "false" for `%s` when passed a boolean: https://github.com/zcash/zcash/blob/27ffefed1291f1474e15149e6992653d1c3669ee/src/tinyformat.h#L737-L742